packages/app-store/larkcalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,426 @@
+      };
+    } catch (error) {
+      this.log.error(error);
+      await this.deleteEvent(eventId, event, calendarId);
+      throw error;
+    }
+  }
+
+  private createAttendees = async (event: CalendarEvent, eventId: string) => {
+    const calendarId = event.destinationCalendar?.externalId;
+    if (!calendarId) {
+      throw new Error("no calendar id");
+    }
+    const accessToken = await this.auth.getToken();
+    const attendeeResponse = await fetch(
+      `https://${LARK_HOST}/open-apis/calendar/v4/calendars/${calendarId}/events/${eventId}/attendees/create_attendees`,
+      {
+        method: "POST",
+        headers: {
+          Authorization: "Bearer " + accessToken,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          attendees: this.translateAttendees(event),
+          need_notification: false,
+        }),
+      }
+    );
+
+    const data = await handleLarkError<CreateAttendeesResp>(attendeeResponse, this.log);
+    return data;
+  };
+
+  /**
+   * @param uid
+   * @param event
+   * @returns
+   */
+  async updateEvent(uid: string, event: CalendarEvent, externalCalendarId?: string): Promise<any> {
+    const eventId = uid;
+    let eventRespData;
+    const calendarId = externalCalendarId || event.destinationCalendar?.externalId;
+    if (!calendarId) {
+      throw new Error("no calendar id");
+    }
+    try {
+      const accessToken = await this.auth.getToken();
+      const eventResponse = await fetch(
+        `https://${LARK_HOST}/open-apis/calendar/v4/calendars/${calendarId}/events/${eventId}/patch_event`,
+        {
+          method: "PATCH",
+          headers: {
+            Authorization: "Bearer " + accessToken,
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(this.translateEvent(event)),
+        }
+      );
+      eventRespData = await handleLarkError<CreateEventResp>(eventResponse, this.log);
+    } catch (error) {
+      this.log.error(error);
+      throw error;
+    }
+
+    try {
+      // Since attendees cannot be changed any more, updateAttendees is not needed
+      // await this.updateAttendees(event, eventId);
+      return {
+        ...eventRespData,
+        uid: eventRespData.data.event.event_id as string,
+        id: eventRespData.data.event.event_id as string,
+        type: "lark_calendar",
+        password: "",
+        url: "",
+        additionalInfo: {},
+      };
+    } catch (error) {
+      this.log.error(error);
+      await this.deleteEvent(eventId, event);
+      throw error;
+    }
+  }
+
+  // Since attendees cannot be changed any more, updateAttendees is not needed
+  // private updateAttendees = async (event: CalendarEvent, eventId: string) => {
+  //   const calendarId = event.destinationCalendar?.externalId;
+  //   if (!calendarId) {
+  //     throw new Error("no calendar id");
+  //   }
+  //   return this.createAttendees(event, eventId);
+  // };
```

Comment:
If this is not needed. Then let's clean this up

Reply:
ok